src/service/viewer-impl.js

```diff
@@ -445,7 +452,40 @@ export class Viewer {
    * @return {!VisibilityState}
    */
   getVisibilityState() {
-    return this.visibilityState_;
+    // The viewer has informed us that the doc is no longer active, meaning
+    // the viewer closed or it swiped to another doc.
+    if (this.visibilityState_ === VisibilityState.UNLOADED ||
+        this.visibilityState_ === VisibilityState.HIDDEN) {
```

Comment:
> translate two values into one here

Are you talking about `UNLOADED || HIDDEN` both return `UNLOADED`? It's because the viewer sends `HIDDEN` down, which is from the point of view of the AMP doc is the "unloaded" state. I'll be filing an internal bug for the viewer to send `UNLOADED` instead.